app/helpers/service_provider_helper.rb

```diff
@@ -128,4 +127,54 @@ def formatted_config_hash(sp_json)
     end.to_h
   end
   # rubocop:enable Layout/LineLength
+
+  def map_config_attributes(sp_hash, sp_id, agency = {name => '', id => nil})
+    base_hash = {
+      'agency_id' => agency['id'],
+      'friendly_name' => sp_hash['friendly_name'],
+      'agency' => agency['name'],
+      'logo' => '<REPLACE_ME.png>',
+      'certs' => '<REPLACE_ME>',
+      'return_to_sp_url' => sp_hash['return_to_sp_url'],
+      'redirect_uris' => sp_hash['redirect_uris'],
+      'ial' => sp_hash['ial'],
+      'attribute_bundle' => sp_hash['attribute_bundle'],
+      'restrict_to_deploy_env' => 'prod',
+      'protocol' => sp_hash['protocol'],
+      'help_text' => sp_hash['help_text'],
+      'app_id' => sp_id,
+      'launch_date' => '<REPLACE_ME>',
+      'iaa' => '<REPLACE_ME>',
+      'iaa_start_date' => '<REPLACE_ME>',
+      'iaa_end_date' => '<REPLACE_ME>',
+    }
+    hash_with_ial_attr = add_IAL_attribute(base_hash, sp_hash['failure_to_proof_url'])
+    if hash_with_ial_attr['protocol'] == 'saml'
+      add_saml_attributes(hash_with_ial_attr)
+    else
+      add_pkce_atttribute(hash_with_ial_attr)
+    end
+  end
+
+  # rubocop:disable Layout/LineLength
```

Comment:
**suggestion:** I generally like overriding rubocop specifically on the lines that violate the linter, so I'd recommend wrapping line 163 instead of the whole method.